app/editor/src/features/reports/ReportInstancePreview.tsx

```diff
@@ -1,35 +1,66 @@
 import React from 'react';
+import { FaPaperPlane } from 'react-icons/fa6';
 import { useParams } from 'react-router-dom';
-import { useReportInstances } from 'store/hooks';
-import { Col, IReportResultModel, Loading, Show } from 'tno-core';
+import { useApp, useReportInstances, useReports } from 'store/hooks';
+import {
+  Button,
+  ButtonVariant,
+  Col,
+  IReportModel,
+  IReportResultModel,
+  Loading,
+  Show,
+} from 'tno-core';
 
 import * as styled from './styled';
 
 const ReportInstancePreview: React.FC = () => {
+  const [{ getReport }] = useReports();
   const [{ viewReportInstance }] = useReportInstances();
   const { id } = useParams();
-  const reportId = parseInt(id ?? '');
+  const instanceId = parseInt(id ?? '');
+  const [{ userInfo }] = useApp();
 
   const [isLoading, setIsLoading] = React.useState(true);
   const [view, setView] = React.useState<IReportResultModel | undefined>();
+  const [report, setReport] = React.useState<IReportModel>();
 
   const handlePreviewReport = React.useCallback(
-    async (reportId: number) => {
+    async (instanceId: number) => {
       try {
         setIsLoading(true);
-        const response = await viewReportInstance(reportId);
+        const response = await viewReportInstance(instanceId);
+        const report = await getReport(response.reportId);
         setView(response);
+        setReport(report);
       } catch {
       } finally {
         setIsLoading(false);
       }
     },
-    [viewReportInstance],
+    [getReport, viewReportInstance],
+  );
+
+  const prepareEmail = React.useCallback(
+    (to: string, report: IReportModel, email: IReportResultModel) => {
+      const subscribers = report.subscribers
+        .filter((s) => s.isSubscribed)
+        .map((s) => (s.user?.preferredEmail ? s.user?.preferredEmail : s.user?.email))
+        .filter((s) => s);
+      const htmlBlob = new Blob([email.body], { type: 'text/html' });
+      const textBlob = new Blob([email.body], { type: 'text/plain' });
+      const clip = new ClipboardItem({ 'text/html': htmlBlob, 'text/plain': textBlob });
+      navigator.clipboard.write([clip]);
+      window.location.href = `mailto:${to}?bcc=${subscribers.join('; ')}&subject=${
```

Comment:
Regrettably we can't automatically paste the body in because it strips the formatting.